src/api/procedures/__tests__/modifyAllowance.ts

```diff
@@ -0,0 +1,183 @@
+import { AccountId, Balance } from '@polkadot/types/interfaces';
+import BigNumber from 'bignumber.js';
+import sinon from 'sinon';
+
+import { getAuthorization, prepareModifyAllowance } from '~/api/procedures/modifyAllowance';
+import { Context, ModifyAllowanceParams, Procedure, Subsidy } from '~/internal';
+import { TxTags } from '~/polkadot';
+import { dsMockUtils, entityMockUtils, procedureMockUtils } from '~/testUtils/mocks';
+import { Mocked } from '~/testUtils/types';
+import { AllowanceOperation } from '~/types/internal';
+import * as utilsConversionModule from '~/utils/conversion';
+
+jest.mock(
+  '~/api/entities/Account',
+  require('~/testUtils/mocks/entities').mockAccountModule('~/api/entities/Account')
+);
+
+jest.mock(
+  '~/api/entities/Subsidy',
+  require('~/testUtils/mocks/entities').mockSubsidyModule('~/api/entities/Subsidy')
+);
+
+describe('modifyAllowance procedure', () => {
+  let mockContext: Mocked<Context>;
+  let subsidy: Subsidy;
+  let stringToAccountIdStub: sinon.SinonStub<[string, Context], AccountId>;
+  let bigNumberToBalanceStub: sinon.SinonStub<
+    [BigNumber, Context, (boolean | undefined)?],
+    Balance
+  >;
+  let args: ModifyAllowanceParams;
+  let allowance: BigNumber;
+  let rawBeneficiaryAccountId: AccountId;
+  let rawAllowance: Balance;
+
+  let increasePolyxLimitTransaction: sinon.SinonStub;
+
+  let proc: Procedure<ModifyAllowanceParams, void>;
+
+  beforeAll(() => {
+    dsMockUtils.initMocks();
+    procedureMockUtils.initMocks();
+    entityMockUtils.initMocks({
+      subsidyOptions: {
+        getAllowance: new BigNumber(100),
+      },
+    });
+
+    stringToAccountIdStub = sinon.stub(utilsConversionModule, 'stringToAccountId');
+    bigNumberToBalanceStub = sinon.stub(utilsConversionModule, 'bigNumberToBalance');
+  });
+
+  beforeEach(() => {
+    mockContext = dsMockUtils.getContextInstance();
+
+    subsidy = entityMockUtils.getSubsidyInstance();
+    allowance = new BigNumber(50);
+    args = { subsidy, operation: AllowanceOperation.Set, allowance };
+
+    rawBeneficiaryAccountId = dsMockUtils.createMockAccountId('beneficiary');
+    stringToAccountIdStub.withArgs('beneficiary', mockContext).returns(rawBeneficiaryAccountId);
+
+    rawAllowance = dsMockUtils.createMockBalance(allowance);
+    bigNumberToBalanceStub.withArgs(allowance, mockContext).returns(rawAllowance);
+
+    increasePolyxLimitTransaction = dsMockUtils.createTxStub('relayer', 'increasePolyxLimit');
+
+    proc = procedureMockUtils.getInstance<ModifyAllowanceParams, void>(mockContext);
+  });
+
+  afterEach(() => {
+    entityMockUtils.reset();
+    procedureMockUtils.reset();
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    procedureMockUtils.cleanup();
+    dsMockUtils.cleanup();
+  });
+
+  test('should throw an error if the Subsidy does not exist', () =>
```

Comment:
```suggestion
  it('should throw an error if the Subsidy does not exist', () =>
```

As well as others in this file